src/QirRuntime/lib/QIR/arrays.cpp

```diff
@@ -55,6 +76,11 @@ QirArray::QirArray(int64_t qubits_count)
     , ownsQubits(true)
     , refCount(1)
 {
+    if (g_context != nullptr && g_context->trackAllocatedObjects)
+    {
+        g_context->allocationsTracker->OnAllocate(this);
+    }
+

```

Comment:
Minor:
Just in case you may want to consider placing this to the _very end_ of the constructors (such that only the _successfully_ and _fully_ constructed instances are added to the `g_context->allocationsTracker`). 

Otherwise the remaining part of the constructor can fail (e.g. during the allocation `QUBIT** qbuffer = new QUBIT*[count]` throw an exception), and the `this->buffer` will stay *un*initialized. And if you break in the debugger upon such a failure and analyze the latest instance in `g_context->allocationsTracker`, then that instance will have a misleading `->buffer`.